File: RxJava_Exam/Java/app/src/main/java/com/example/dgsw/rxjava_exam/TodoDao.java
package com.example.dgsw.rxjava_exam;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;
import java.util.Observable;

import io.reactivex.Completable;
import io.reactivex.Flowable;

@Dao
public interface TodoDao {
    @Query("SELECT * FROM Todo")
    LiveData<List<Todo>> getAll();

    @Insert
    Completable insert(Todo todo);

    @Update
    Completable update(Todo todo);

    @Delete
    Completable delete(Todo todo);
}

File: RxJava_Exam/Kotlin/app/src/main/java/com/example/dgsw/rxjava_exam/TodoDao.kt
package com.example.dgsw.rxjava_exam

import androidx.lifecycle.LiveData
import androidx.room.*
import io.reactivex.Completable
import io.reactivex.Flowable
import io.reactivex.Observable

@Dao
interface TodoDao {
    @Query("SELECT * FROM Todo")
    fun getAll(): Observable<List<Todo>>

    @Insert
    fun insert(todo : Todo) : Completable

    @Update
    fun update(todo : Todo) : Completable

    @Delete
    fun delete(todo : Todo) : Completable
}